packages/engine/src/plugin/Plugin.ts

```diff
@@ -0,0 +1,25 @@
+import { ParamWithInfo } from "@store/selectors/getParamWithInfo";
+
+/**
+ * Class type for a Plugin.
+ */
+export interface IPlugin {
+    // new(engine: HedronEngine, useEngineStore: UseEngineStore): any;
+
+    /**
+     * The name of the plugin.
+     */
+    name: string;
+
+    /**
+     * The description of the plugin.
+     */
+    description: string;
+
+    /**
+     * Gets the selected parameter view.
+     * @param param - The parameter with info.
+     * @returns The JSX element or undefined if this plugin does not have a visual element.
+     */
+    getSelectedParamView: (param: ParamWithInfo) => JSX.Element | undefined;
```

Comment:
I'm thinking starting with 2 optional functions for plugins, one for the selected param display, and one as a 'main' display (ie clock under the render view)

Reply:
I think eventually we'll want plugins to have types. So an `input` type might have something like `paramView` and a `settingsView`. Perhaps something more general might be a `panel` type that has a `panelView` and also `settingsView`. Not sure yet! For now I'd say just keep it simple and have only one view for inputs, unless you already have an idea for what `main` might be?

Reply:
In terms of showing MIDI devices, do you see that as part of the same plugin? I'd suggest that the MIDI device manager is just a permanent core feature (can still live as a separate package) that all plugins can talk to?